File: src/test/java/tests/de/espend/idea/vuejs/index/visitor/AppEntryPsiRecursiveElementVisitorTest.java
package tests.de.espend.idea.vuejs.index.visitor;

import com.intellij.psi.PsiFile;
import de.espend.idea.vuejs.index.visitor.AppEntryPsiRecursiveElementVisitor;
import tests.de.espend.idea.vuejs.VueJsLightJavaCodeInsightFixtureTestCase;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
public class AppEntryPsiRecursiveElementVisitorTest extends VueJsLightJavaCodeInsightFixtureTestCase {
    public String getTestDataPath() {
        return "src/test/java/tests/de/espend/idea/vuejs/index/fixtures";
    }

    public void testCreateapp() {
        PsiFile psiFile = myFixture.configureByFile("appentry-createapp.js");

        Map<String, List<String>> map = new HashMap<>();
        psiFile.acceptChildren(new AppEntryPsiRecursiveElementVisitor(map));

        List<String> app = map.get("App");

        assertEquals("./App.vue", app.get(0));
        assertEquals("createApp", app.get(1));
    }

    public void testCreateappV2() {
        PsiFile psiFile = myFixture.configureByFile("appentry-createelement-v2.ts");

        Map<String, List<String>> map = new HashMap<>();
        psiFile.acceptChildren(new AppEntryPsiRecursiveElementVisitor(map));

        List<String> app = map.get("App");

        assertEquals("./App.vue", app.get(0));
        assertEquals("createElement", app.get(1));
    }
}

File: src/test/kotlin/tests/de/espend/idea/vuejs/index/visitor/AppEntryPsiRecursiveElementVisitorTest.kt
package tests.de.espend.idea.vuejs.index.visitor

import de.espend.idea.vuejs.index.visitor.AppEntryPsiRecursiveElementVisitor
import tests.de.espend.idea.vuejs.VueJsLightJavaCodeInsightFixtureTestCase

class AppEntryPsiRecursiveElementVisitorTest : VueJsLightJavaCodeInsightFixtureTestCase() {
    override fun getTestDataPath() = "src/test/kotlin/tests/de/espend/idea/vuejs/index/fixtures"

    fun testCreateapp() {
        val psiFile = myFixture.configureByFile("appentry-createapp.js")

        val map = mutableMapOf<String, List<String>>()
        psiFile.acceptChildren(AppEntryPsiRecursiveElementVisitor(map))

        val app = map["App"]
        assertEquals("./App.vue", app!![0])
        assertEquals("createApp", app[1])
    }

    fun testCreateappV2() {
        val psiFile = myFixture.configureByFile("appentry-createelement-v2.ts")

        val map = mutableMapOf<String, List<String>>()
        psiFile.acceptChildren(AppEntryPsiRecursiveElementVisitor(map))

        val app = map["App"]
        assertEquals("./App.vue", app!![0])
        assertEquals("createElement", app[1])
    }

    fun testCreateappV2New() {
        val psiFile = myFixture.configureByFile("appentry-mount-v2.js")

        val map = mutableMapOf<String, List<String>>()
        psiFile.acceptChildren(AppEntryPsiRecursiveElementVisitor(map))

        val app = map["App"]
        assertEquals("./App.vue", app!![0])
        assertEquals("\$mount", app[1])
    }
}
